cmd/argo/commands/node.go

```diff
@@ -0,0 +1,101 @@
+package commands
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/argoproj/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/argoproj/argo/cmd/argo/commands/client"
+	workflowpkg "github.com/argoproj/argo/pkg/apiclient/workflow"
+)
+
+type setOps struct {
+	message           string   // --message
+	phase             string   // --phase
+	outputParameters  []string // --output-parameters
+	nodeFieldSelector string   // --node-field-selector
+}
+
+func NewNodeCommand() *cobra.Command {
+	var (
+		setArgs setOps
+	)
+
+	var command = &cobra.Command{
+		Use:   "node ACTION WORKFLOW FLAGS",
+		Short: "perform action on a node in a workflow",
+		Example: `# Set outputs to a node within a workflow:
+
+  argo node set my-wf --output-parameter parameter-name="Hello, world!" --node-field-selector displayName=approve
+
+# Set the message of a node within a workflow:
+
+  argo node set my-wf --message "We did it!"" --node-field-selector displayName=approve
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+
+			if len(args) < 1 {
+				cmd.HelpFunc()(cmd, args)
+			}
+
+			if args[0] != "set" {
```

Comment:
Do you think we can have a separate command for a `set` similar like a template create instead of args check? This will allow for future enhancements.

Reply:
I'm sorry, I don't think I understand what you mean. Could you provide an example?

Reply:
`set` is a subcommand of `node` command. you can create `set` separate command under node instead of using as args.  In the future, if we want another commend, we can easily add another sub-command. you can refer the existing command implementation example `template`